paper.tex

```diff
@@ -1535,11 +1535,43 @@ \subsection{Extensibility: GPU acceleration}
 on GPU. In the second, we demonstrate some emerging technology 
 in which Zarr data can be decompressed directly in GPU memory. 
 In both cases we demonstrate on chromosome 22 of 
-1000 Genomes WGS data~\citep{byrska2022high}. 
+1000 Genomes WGS data~\citep{byrska2022high}. This consists
+of 26,504 variants and 2504 samples [and more details about the zarr].
+Benchmarks in this section were performed on a
+machine with an Intel Xeon Silver 4216 processor,
+Nvidia A6000 RTX GPU and NVMe SSD.
+
+We first illustrate the ease with which computations can be 
+be peformed on the GPU, with little or no preprocessing of the 
+data. To do this we reimplemented the af-dist calculation
+described above using CuPy~\citep{cupy}, a GPU accelerated library
+of core operations for scientific computing.
+All of the operations required to compute  af-dist are 
+performed on the GPU, mostly with minimal changes required 
+(e.g.\ replacing \texttt{np.digitize} with \texttt{cp.digitize}).
+We perform the core operation of genotype classification using the
+CuPy JIT compiler to generate a custom CUDA kernel, which 
+is also very similar to the Numba compiled function used in other
+benchmarks. In the simplest version, we iterate over the chunks
+in the \texttt{call\_genotype} array, decompress each to a local
+NumPy array and then copying the data to GPU memory by creating 
+the corresponding CuPy array. For this version, the total time
+in the genotype classification kernel was 0.045s, with an 
+overall data transfer time of 0.19s (we discount decompression
+time here; see next paragraph). For comparison, 
+our standard Numba compiled classification function required 
+a total of 1.44s running on a single CPU core.
+Another approach in which we load the entire 
+\texttt{call\_genotype} array into GPU memory at once (possible
+here as the $96514 \times 2504 \times 2$ array requires 
+460MiB) and use
+more sophisticated kernel in which we map GPU resources to
```

Comment:
Should this be 'and use**s a** more sophisticated kernel in which we map GPU resources to subsets of the array in two dimensions, **which** reduces **the** transfer time...'?

Reply:
corrected in upcoming updates